nexus/db-queries/src/db/datastore/instance.rs

```diff
@@ -385,6 +389,44 @@ impl DataStore {
         Ok((instance_updated, vmm_updated))
     }
 
+    pub async fn instance_and_vmm_list_by_sled_agent(
+        &self,
+        opctx: &OpContext,
+        pagparams: &DataPageParams<'_, Uuid>,
+    ) -> ListResultVec<(Sled, Instance, Vmm, Project)> {
+        use crate::db::schema::{
+            instance::dsl as instance_dsl, project::dsl as project_dsl,
+            sled::dsl as sled_dsl, vmm::dsl as vmm_dsl,
+        };
+        opctx.authorize(authz::Action::Read, &authz::INVENTORY).await?;
```

Comment:
I'm not necessarily suggesting a change, just pointing out something quirky here.

In `nexus/db-queries/src/authz/api_resources.rs`, INVENTORY is labelled as "Synthetic resource used for modeling access to low-level hardware inventory data". Does that fit the bill here?

Should this be authorizing read access to `FLEET`?

I'm not necessarily pushing for a change here because within the context of this change, I think the behavior will be largely the same, and "how we bucket our authz resources" also seems a little arbitrary

Reply:
Ah, yeah, this should probably be `FLEET`,  I think this is residue from an earlier version of this change where we started by reading sleds from the inventory. I'll update it!